File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/IChatCommand.java
package me.nobaboy.nobaaddons.features.chatcommands;

import java.util.Collections;
import java.util.List;

public interface IChatCommand {
    String name();

    default String usage() {
        return name();
    }

    void run(ChatContext ctx);

    default boolean isEnabled() {
        return true;
    }

    default List<String> aliases() {
        return Collections.emptyList();
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/IChatCommand.kt
package me.nobaboy.nobaaddons.features.chatcommands

interface IChatCommand {
    val name: String

    val usage: String
        get() = name

    fun run(ctx: ChatContext)

    val isEnabled: Boolean
        get() = true

    val aliases: MutableList<String>
        get() = mutableListOf()

    val bypassCooldown: Boolean
        get() = false
}